scripts/cl_acquisition.py

```diff
@@ -60,120 +66,268 @@ def __init__(self):
+        self.escan.dwellTime.value = self.escan.dwellTime.clip(exp)
+
+        # Configure the SEM to spot mode
+        self.escan.resolution.value = (1, 1)
+        self.escan.scale.value = (1, 1)  # to be certain we can easily move the spot anywhere
+        spot_pos = self.get_spot_positions()
+        logging.debug("Generating %dx%d spots for %g s",
+                      spot_pos.shape[1], spot_pos.shape[0], exp)
+
+        sem_data = []
+        opt_data = []
+
+        # Loop on the spot positions
+        for px_idx in numpy.ndindex(*self.repetition[::-1]):  # last dim (X) iterates first
+            trans = tuple(spot_pos[px_idx])  # spot position
+
+            self.escan.translation.value = trans
+
+            # acquire sem (constantly)
+            self.start_sem_acquisition()
+            # acquire optical data
+            opt_img = self.get_optical_image()
+            opt_data.append(opt_img)
+
+            # Finish the sem acquisition
+            sed_img = self.complete_sem_acquisition()
+            sem_data.append(sed_img)
+
+        # build final arrays
+        sem_da = self.assemble_sem_data(sem_data)
+        opt_da = self.assemble_opt_data(opt_data, sem_da)
+
+        return sem_da, opt_da
+
+
+def main(args):
+    """
+    Handles the command line arguments
+    args is the list of arguments passed
+    return (int): value to return to the OS as program exit code
+    """
+
+    # arguments handling
+    parser = argparse.ArgumentParser(description="CL spectrum acquisition")
+
+    parser.add_argument("--roi", dest="roi", required=True,
+                        help="e-beam ROI positions (left top bottom right, relative to the SEM "
+                             "field of view)")
+    parser.add_argument("--repetition", dest="repetition", required=True,
+                        help="total of points in X and Y")
+    parser.add_argument("--output", "-o", dest="filename", required=True,
+                        help="Output filename")
+
+    options = parser.parse_args(args[1:])
+
+    try:
+        if "." not in options.filename[-5:]:
+            raise ValueError("Output argument must contain extension, "
+                             "but got '%s'" % (options.filename,))
+
+        roi = conversion.reproduce_typed_value([1.0], options.roi)
+        if not all(0 <= r <= 1 for r in roi):
+            raise ValueError("roi values must be between 0 and 1")
+
+        repetition = conversion.reproduce_typed_value([1], options.repetition)
+        if not all(1 <= r for r in repetition):
+            raise ValueError("repetition values must be >= 1")
+
+        acquirer = Acquirer(repetition, roi)
+        sed_data, spect_data = acquirer.acquire_cube()
+
+        exporter = dataio.find_fittest_converter(options.filename)
+        exporter.export(options.filename, [sed_data, spect_data])
+    except Exception:
+        logging.exception("Unexpected error while performing action.")
```

Comment:
```suggestion
    except Exception as e:
        logging.exception(f"Unexpected error while performing action. Error: {e}")
```

Reply:
`logging.exception` is special: it automatically prints the error, and the backtrace. So no need to explicitly include the error in the message.